docs/metrics/custom-metrics.md

```diff
@@ -0,0 +1,122 @@
+* HTTP request method: GET or POST
+* Format of HTTP parameters and/or JSON body used while querying them
+* Format of the JSON response returned by the provider
+* The logic used by Iter8 to extract the metric value from the JSON response
+
+The examples in this document focus on Prometheus, NewRelic, Sysdig, and Elastic. However, the principles illustrated here will enable you to use metrics from any provider in experiments.
+
+## Metrics with/without auth
+
+> **Note:** Metrics are defined by you, the **Iter8 end-user**.
+
+=== "Prometheus"
+
+    Prometheus does not support any authentication mechanism *out-of-the-box*. However,
+    Prometheus can be setup in conjunction with a reverse proxy, which in turn can support HTTP request authentication, as described [here](https://prometheus.io/docs/guides/basic-auth/).
+
+    === "No Authentication"
+        The following is an example of an Iter8 metric with Prometheus as the provider. This example assumes that Prometheus can be queried by Iter8 without any authentication.
+
+        ```yaml linenums="1"
+        url: http://127.0.0.1:9090/api/v1/query
+        provider: istio
+        method: GET
+        metrics:
+        - name: request-count
+          type: counter
+          description: |
+            Number of requests
+          params:
+          - name: query
+            value: |
+              sum(last_over_time(istio_requests_total{
+                  reporter="source",    
+                {{- if .destination_workload }}
+                  destination_workload="{{.destination_workload}}",
+                {{- end }}
+                {{- if .destination_workload_namespace }}
+                  destination_workload_namespace="{{.destination_workload_namespace}}",
+                {{- end }}
+              }[{{.elapsedTimeSeconds}}s])) or on() vector(0)
+          jqExpression: .data.result[0].value[1]
+        ```
+
+    ??? hint "Brief explanation of the `request-count` metric"
+
+        1. The HTTP query used by Iter8 contains a single query parameter named `query` as [required by Prometheus](https://prometheus.io/docs/prometheus/latest/querying/api/). The value of this parameter is derived by [substituting the placeholders](#placeholder-substitution) in the value string.
+        2. The `url` field provides the URL of the Prometheus service.
+        3. The `method` field provides the HTTP method, in this case `GET`.
+        4. The `jqExpression` enables Iter8 to extract the metric value from the JSON response returned by Prometheus.
+
+## Placeholder substitution
+
+> **Note:** This step is automated by **Iter8**.
+
+=== "Prometheus"
+  The placeholder `elapsedTimeSeconds` is substituted based on the start of the experiment or `startingTime`, if provided in the CLI. If `startingTime` is provided, then the time should follow [RFC 3339](https://www.rfc-editor.org/rfc/rfc3339) (for example: `2020-02-01T09:44:40Z` or `2020-02-01T09:44:40.954641934Z`). 
```

Comment:
I believe `2020-02-01T09:44:40.954641934Z` is what we're more likely to see from Kubernetes so I included it here as well.